### ISSUE TITLE
differentiate between demos that finished recording and demos that are still being written to

### ISSUE THREAD

**Issue by ChillerDragon** [ddnet/ddnet#9100]
Just looking at the filesystem or the demo browser list it is hard to tell if a demo is still being written to.
I propose a .tmp suffix should be used for demos that are being recorded and it should be removed when the recording finished.
And the clients demo browser should not display those files to avoid playing back demos that are being written to at the same time.

**Comment by def-** [comment]
I agree that we should somehow mark unfinished demo files, maybe a .tmp.demo suffix is enough? I have previously opened a demo that was still being recorded in a second client to create a clip while I was still playing, by using a .demo.tmp file ending you'd lose that ability.

**Comment by Robyt3** [comment]
See #7349 and #7790. I can revive the PR if we want it now.

**Comment by ChillerDragon** [comment]
I would prefer .demo.tmp over .tmp.demo we can still let the client index those in the demo browser but then marked as incomplete demo files. A file ending in .demo should be fully written and correct.

Makes working with the files easier. For example moving all old demos to another drive without corrupting active demos:

```
mv *.demo /mnt/data0
```

**Comment by jxsl13** [comment]
> I would prefer .demo.tmp over .tmp.demo we can still let the client index those in the demo browser but then marked as incomplete demo files. A file ending in .demo should be fully written and correct.
> 
> Makes working with the files easier. For example moving all old demos to another drive without corrupting active demos:
> 
> ```
> mv *.demo /mnt/data0
> ```

agreed
